mmpose/datasets/pipelines/gesture_transform.py

```diff
@@ -0,0 +1,400 @@
+            start_h, start_w = (height - self.length) // 2, (width -
+                                                             self.length) // 2
+            video = mmcv.imcrop(
+                video,
+                np.array((start_w, start_h, start_w + self.length - 1,
+                          start_h + self.length - 1)))
+            results['video'][i] = video.reshape(self.length, self.length, -1,
+                                                num_frames).transpose(
+                                                    3, 0, 1, 2)
+            results['width'][i], results['height'][
+                i] = self.length, self.length
+        return results
+
+
+@PIPELINES.register_module()
+class ModalWiseChannelProcess:
+    """Video channel processing according to modality.
+
+    Required keys: 'video', 'modality'.
+
+    Modified keys: 'video'.
+    """
+
+    def __init__(self):
+        pass
+
+    def __call__(self, results):
+        """Implement channel processing for video array."""
+        for i, modal in enumerate(results['modality']):
+            if modal == 'rgb':
+                results['video'][i] = results['video'][i][..., ::-1]
+            elif modal == 'depth':
+                if results['video'][i].ndim == 4:
+                    results['video'][i] = results['video'][i][..., :1]
+                elif results['video'][i].ndim == 3:
+                    results['video'][i] = results['video'][i][..., None]
+            elif modal == 'flow':
+                results['video'][i] = results['video'][i][..., :2]
+            else:
+                raise ValueError(f'modality {modal} is invalid.')
+        return results
+
+
+@PIPELINES.register_module()
+class MultiModalVideoToTensor:
+    """Data processing by converting video arrays to pytorch tensors.
+
+    Required keys: 'video', 'modality'.
+
+    Modified keys: 'video'.
+    """
+
+    def __init__(self):
+        pass
+
+    def __call__(self, results):
+        """Implement data processing similar to ToTensor."""
+        for i, modal in enumerate(results['modality']):
+            video = results['video'][i].transpose(3, 0, 1, 2)
+            results['video'][i] = torch.tensor(
+                np.ascontiguousarray(video), dtype=torch.float) / 255.0
+        return results
+
+
+@PIPELINES.register_module()
+class VideoNormalizeTensor:
+    """Data processing by converting video arrays to pytorch tensors.
+
+    Required keys: 'video', 'modality'.
+
+    Modified keys: 'video'.
+    """
+
+    def __init__(self, mean, std):
+        self.mean = torch.tensor(mean)
+        self.std = torch.tensor(std)
+
+    def __call__(self, results):
+        """Implement data normalization."""
+        for i, modal in enumerate(results['modality']):
+            if modal == 'rgb':
+                video = results['video'][i]
+                dim = video.ndim - 1
+                video = video.sub(self.mean.view(3, *((1, ) * dim)))
```

Comment:
Can we simply use `-` instead of `sub` ?

Reply:
Ok, both `sub` and `div` are replaced.